state/action.go

```diff
@@ -449,20 +458,23 @@ func (a *action) removeAndLogBuildTxn(finalStatus ActionStatus, results map[stri
 					}
 				}
 				updateOperationOp = &txn.Op{
-					C:      operationsC,
-					Id:     a.st.docID(parentOperation.Id()),
-					Assert: assertNotComplete,
+					C:  operationsC,
+					Id: a.st.docID(parentOperation.Id()),
+					Assert: append(assertNotComplete,
+						bson.DocElem{"complete-task-count", bson.D{{"$eq", expectedTaskCount - 1}}}),
 					Update: bson.D{{"$set", bson.D{
 						{"status", finalOperationStatus},
 						{"completed", completedTime},
-						{"complete-task-count", numComplete + 1},
+						{"complete-task-count", expectedTaskCount},
```

Comment:
```suggestion
						{"spawned-task-count", expectedTaskCount},
```

Reply:
In this case "numComplete + 1" is the same value as "spawnedTaskCount", given that we're completing the operation, I wanted to be clear that complete-task-count was now the same as the spawned-task-count.  The spawned-task-count should never change after the operation is created.

Reply:
The suggestion in this comment and the ones below were for renaming `complete-task-count` to `spawned-task-count` which is the new name for that field after it got renamed. I can't check ATM but I believe this is why the tests are failing.